pkg/cmd/roachtest/tests/restore.go

```diff
@@ -813,6 +813,58 @@ func (tpce tpceRestore) DatabaseName() string {
 	return "tpce"
 }
 
+type tpccRestoreOptions struct {
+	warehouses     int
+	workers        int
+	waitFraction   float64
+	queryTraceFile string
+}
+
+type tpccRestore struct {
+	opts tpccRestoreOptions
+}
+
+func (tpccRestore) getNodes(c cluster.Cluster) (crdbNodes, workloadNode option.NodeListOption) {
+	return c.Range(1, c.Spec().NodeCount-1), c.Node(c.Spec().NodeCount)
+}
+
+func (tpcc tpccRestore) init(
+	ctx context.Context, t test.Test, c cluster.Cluster, sp hardwareSpecs,
+) {
+	crdbNodes, workloadNode := tpcc.getNodes(c)
+	cmd := roachtestutil.NewCommand(`./cockroach workload init tpcc`).
+		MaybeFlag(tpcc.opts.warehouses > 0, "warehouses", tpcc.opts.warehouses).
+		MaybeFlag(tpcc.opts.workers > 0, "workers", tpcc.opts.workers).
+		MaybeFlag(tpcc.opts.waitFraction != 1, "wait", tpcc.opts.waitFraction).
+		Arg(fmt.Sprintf("{pgurl:%d-%d}", crdbNodes[0], crdbNodes[len(crdbNodes)-1]))
+	c.Run(ctx, option.WithNodes(workloadNode), cmd.String())
+}
+
+func (tpcc tpccRestore) run(
+	ctx context.Context, t test.Test, c cluster.Cluster, sp hardwareSpecs,
+) error {
+	crdbNodes, workloadNode := tpcc.getNodes(c)
+	cmd := roachtestutil.NewCommand(`./cockroach workload run tpcc`).
+		MaybeFlag(tpcc.opts.warehouses > 0, "warehouses", tpcc.opts.warehouses).
+		MaybeFlag(tpcc.opts.workers > 0, "workers", tpcc.opts.workers).
+		MaybeFlag(tpcc.opts.waitFraction != 1, "wait", tpcc.opts.waitFraction).
+		MaybeFlag(tpcc.opts.queryTraceFile != "", "query-trace-file", tpcc.opts.queryTraceFile).
+		Arg(fmt.Sprintf("{pgurl:%d-%d}", crdbNodes[0], crdbNodes[len(crdbNodes)-1]))
+	return c.RunE(ctx, option.WithNodes(workloadNode), cmd.String())
+}
+
+func (tpcc tpccRestore) fixtureDir() string {
+	return fmt.Sprintf("tpc-c/warehouses=%d", tpcc.opts.warehouses)
```

Comment:
nit: remove the hyphen

Reply:
Currently the `tpce` path is named `tpc-e`, do we want to keep that consistent?